### ISSUE TITLE
Security of CO15 ciphertext computation

### ISSUE THREAD

**Issue by rozbb** [emp-toolkit/emp-ot#74]
Currently, the [CO15 OT](https://eprint.iacr.org/2015/267.pdf) sender output [uses](https://github.com/emp-toolkit/emp-ot/blob/5ef31873932096e672039bf41bc5ce5ac60f0bae/emp-ot/co.h#L53) the encryption algorithm `E(k, m) = k ⊕ m`. But this is not the algorithm that CO15 use themselves!

CO15 use a quasi-authenticated cipher of the form `E'(k, m) = (k₁ ⊕ m, k₂)` where `k₁` and `k₂` are the two 128-bit halves of `k`. Importantly, they argue that `E'` is _non-committing_ and _robust_ as per definitions 1 and 2 (see screenshot; it's from page 5). On the other hand, `E` is NOT robust: `V_{S,e}` always equals `S` in the definition, since decryption never fails.

Is the choice of the encryption algorithm `E` secure here, then?

<img width="824" alt="Screen Shot 2022-06-09 at 15 11 16" src="https://user-images.githubusercontent.com/752802/172925645-fdedf44e-e98c-486a-87ec-2d5615ce4778.png">

**Comment by wangxiao1254** [comment]
Note that `B[i]` is fed into a KDF (which you can model as a random oracle). Putting it in their language, it is more like $H(k)\xor m$.

**Comment by rozbb** [comment]
Yes, this is in keeping with CO15's definition of $k^i_j = H_{(S, R^i)}(whatever)$

**Comment by rozbb** [comment]
Ah, interesting. What's PVW?

**Comment by wangxiao1254** [comment]
https://eprint.iacr.org/2007/348

**Comment by rozbb** [comment]
Thanks. So to recap: is this simplified encryption function `E` safe to use? I ask because I'm implementing CO15 myself and it'd be nice if the ciphertext was 50% smaller

**Comment by wangxiao1254** [comment]
If safe means no attack, I think so.
If safe means simulation security, even the original version of CO15 is not. You need to do either PVW or a patched version of CO15. See e.g., https://eprint.iacr.org/2018/499

**Comment by wangxiao1254** [comment]
In the latest branch, we have implemented CSW and PVW.
